src/components/MainInfo/utils.tsx

```diff
@@ -0,0 +1,204 @@
+import { GetAvailibleDataFunction, GetMainInfoStructureFunction } from '../../types/types';
+import { uuidExtract } from '../../utils/extract';
+import { currentMargin, headerHeight, tabsHeight } from '../DesignSystem/DesignSystem';
+
+/** @todo fixme, i want to have responsive calculated sizes */
+export const scrollMarginTop = headerHeight + tabsHeight + currentMargin;
+
+export const getAvailibleData: GetAvailibleDataFunction = (mainInfoStructure) => {
+  return mainInfoStructure.map((article) => {
+    return article.fields
+      ?.map((column) => {
+        if (typeof column.value === 'boolean') return true;
+        return !!column.value;
+      })
+      .some((value: any) => value !== false);
+  });
+};
```

Comment:
If I understand, this function tells whether at least one field of an 'article' has a value. Then, in the `MainInfo` component, we get `dataStructure` and `availableData`. While looping over `dataStructure` in the JSX, we check `availableData` to check if a row should be rendered.

Would be possible to have a function that returns the data that should be rendered by components like `MainInfo` and then no additional checks have to be done in the component itself?

I feel that now one has to look at multiple different places to understand what is going to be rendered and maybe this can be simplified or contained a bit more?

Reply:
I have changed the `getAvalibleData` function, and put in inside the `Content` component (previously `MainInfo`. The data structure is passed to the `Content` component and will validate the data an make articles an columns.